src/v/storage/compacted_index.h

```diff
@@ -69,10 +69,15 @@ struct compacted_index {
         // 0 - initial version
         // 1 - introduced a key being a tuple of batch_type and the key content
         //  of footer
-        static constexpr int8_t current_version = 1;
-
-        uint32_t size{0};
-        uint32_t keys{0};
+        // 2 - 64-bit size and keys fields
+        static constexpr int8_t current_version = 2;
+
+        uint64_t size{0};
+        uint64_t keys{0};
+        // must be kept for backwards compatibility with pre-version 2 code
+        // (that allows using an index with a version greater than current).
+        uint32_t size_deprecated{0};
```

Comment:
Is the idea of the _deprecated fields to make indices written by latest code readable by older code?  If so, I would have thought they would have to come first?

Reply:
> Is the idea of the _deprecated fields to make indices written by latest code readable by older code?

yes exactly, older code won't bail out if it sees version 2, so we need to account for that.

>  If so, I would have thought they would have to come first?

It is a footer, so old code will read the last 17 bytes of the file and try to interpret that as a footer. So they are first from the end :)

Reply:
> It is a footer, so old code will read the last 17 bytes of the file and try to interpret that as a footer. So they are first from the end :)

Aaaah :facepalm: